### DISCUSSION TITLE
Noisy Color Attribute Output on Inpainted 3D Mesh

### DISCUSSION THREAD

**Discussion by magda-kk** [thygate/stable-diffusion-webui-depthmap-script#181] [Q&A] [answered]
When importing my generated inpainted 3D Mesh into Blender, and trying to apply the texture, it turns out really noisy (just the borders look normal) This has happened with every generation so far. The video demos seem to work, so I'm not sure if it is a blender issue (ver 3.4.0). Does anyone know how to fix this?
![image](https://user-images.githubusercontent.com/76952547/231908851-de92a8ba-63fa-4178-803a-bf9e4f0bfe2d.png)

**Comment by thygate** [comment]
If it looks good in the video generation then the mesh is fine.

What if you only connect the color attribute to the BSDF base color input, and the BSDF output to material output surface.

Also check https://github.com/thygate/stable-diffusion-webui-depthmap-script/issues/175#issuecomment-1506124967

**Reply by magda-kk** [reply]
I already connected it to the Principled BSDF and connected that to the output, that has the same noise.
I saw the comment mentioned in a different discussion, but I'm not sure how I can use it? The comment shows someone plugging in an image texture instead of using the color attribute, right? I also tried to project the texture, but I'm not sure what the right camera distance is and as far as I understood the inpainted color attribute should be better, right?

**Reply by thygate** [reply]
was the image larger than 1024 on any side ?

If so, try changing `Max size for generating simple mesh.` in the settings tab to a value larger than your image size, a power of 2, so 2048, 4096, .. etc

**Reply by thygate** [reply]
Yes, I'm not sure what technique was used in that post, I've only used the vertex color attribute, and it should look fine in blender. I'm still on Blender 3.3.1 though ..

**Reply by magda-kk** [reply]
Alright, I will try installing an older version of Blender and see if it works there, thank you!

**Reply by magda-kk** [reply]
No that did not work either :/ the image was larger than that, but I am not generating a simple mesh - does that still affect it? Gonna try it anyways and update here if it worked

**Reply by thygate** [reply]
>Max size for generating simple mesh.

oh wait, my bad, I got things mixed up there, i'm trying to do a lot of different things at once here..

**Reply by thygate** [reply]
If the mesh renders correctly using the generate video in the extension, then it is fine.

Did you perhaps use the legacy OBJ importer ?

![image](https://user-images.githubusercontent.com/96586282/231917671-1d52045b-b814-42b0-9eaa-a724fa554bfd.png)

**Comment by thygate** [comment, marked as answer]
Perhaps try enabling the 'save additional ply model' in depthmap settings, and import the ply file instead of the obj file, does that make a difference ?

**Reply by TheMasterFX** [reply]
Worked for me!

**Reply by thygate** [reply]
That's strange. It must be an issue with the wavefront importer in blender. Make sure you're not using the legacy importer (See https://github.com/thygate/stable-diffusion-webui-depthmap-script/discussions/181#discussioncomment-5610697)

Also a bit ironic, since the wavefront obj standard does not include an option for vertex colors, but the obj format that blender exports does, which is how I implemented it too ..

**Reply by magda-kk** [reply]
Thank you so much, the ply file works!
